### ISSUE TITLE
Problems specifying resource manager and running with CDH

### ISSUE THREAD

**Issue by koverholt** [blaze/rambling#46]
I'm running knit alongside a CDH 5.5 cluster with the default YARN setup. When I run with knit, I get an error when running with the default settings:

```
>>> from knit import Knit
>>> k = Knit(autodetect=True)
>>> cmd = 'date'
>>> app_id = k.start(cmd)
>>> k.status(app_id)

...

ConnectionError: HTTPConnectionPool(host='localhost', port=8088): Max retries exceeded with url: 
/ws/v1/cluster/apps/ (Caused by 
NewConnectionError('<requests.packages.urllib3.connection.HTTPConnection object at 
0x7f66c93b3160>: Failed to establish a new connection: [Errno 111] Connection refused',))
```

But I'm unable to configure the resourcemanager, even though it's falling back to the `localhost` config option:

```
>>> k = Knit(rm='ip-172-31-24-75.ec2.internal')

...

HDFSConfigException: Possible Resource Manager hostname mismatch.  Detected localhost
```

So, a couple of issues:

1) I should be able to override the autodetect settings even if they don't match config files or the fallback option.

2) How can we make this work with CDH 5.5 out of the box, since I don't see any `*HADOOP*` env variables defined on my CDH node. Should I have to define this myself, or can we use another signal or search for the `yarn-site.xml` file?

**Comment by quasiben** [comment]
I think knit can do a better job here and definitely relax the error raising

**Comment by martindurant** [comment]
Fixed with new configuration system #71
